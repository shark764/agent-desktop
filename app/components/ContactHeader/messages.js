/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 * ContactHeader Messages
 *
 * This contains all the text for the ContactHeader component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  newContactBanner: {
    id: 'app.containers.InfoTab.newContact',
    defaultMessage: 'New Customer Record',
  },
  contactEditingBanner: {
    id: 'app.containers.SidePanel.editing',
    defaultMessage: 'Editing Customer Record',
  },
  contactMergeBanner: {
    id: 'app.containers.SidePanel.merging',
    defaultMessage: 'Edit Merged Record',
  },
  edit: {
    id: 'app.containers.InfoTab.edit',
    defaultMessage: 'Edit',
  },
});
