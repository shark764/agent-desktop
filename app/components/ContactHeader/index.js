/* eslint react/prop-types: 0 */
/**
*
* ContactHeader
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';

import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';

import messages from './messages';

function ContactHeader(props) {
  const styles = {
    bannerHeader: {
      backgroundColor: '#DEF8FE',
      width: 'calc(100% + 76px)',
      height: '70px',
      position: 'relative',
      left: '-51px',
      display: 'flex',
      alignItems: 'stretch',
      flexShrink: 0,
    },
    bannerHeaderText: {
      fontWeight: 'bold',
      alignSelf: 'center',
    },
    leftGutter: {
      width: '52px',
    },
    controlHeader: {
      minHeight: '70px',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      flexShrink: '0',
    },
    buttonSet: {
      alignSelf: 'flex-end',
      flexGrow: '0',
      flexShrink: '1',
    },
    leftButton: {
      margin: '0 11px',
    },
    rightButton: {
      float: 'right',
      margin: '0',
      height: '36px',
      width: '60px',
    },
  };

  function getBannerHeader(text) {
    return (
      <div style={styles.bannerHeader}>
        <div style={styles.leftGutter}></div>
        <div style={styles.bannerHeaderText}>
          {text}
        </div>
      </div>
    );
  }

  function getViewControlHeader() {
    return (
      <div style={styles.controlHeader}>
        {
          props.selectedInteraction.interactionId !== 'creating-new-interaction'
          && <div style={styles.buttonSet}>
            <Button id="contact-edit-btn" style={styles.leftButton} onClick={props.editAssignedContact} text={messages.edit} type="secondary" />
            <Button id="contact-search-btn" style={styles.rightButton} onClick={props.setSearching} iconName="search" type="secondary" />
          </div>
        }
      </div>
    );
  }

  switch (props.selectedInteraction.contactAction) {
    case 'view':
      if (props.contactMode === 'editing') {
        return getBannerHeader(<FormattedMessage {...messages.contactEditingBanner} />);
      }
      return getViewControlHeader();
    case 'search':
    default:
      if (props.contactMode === 'editing') {
        if (Object.keys(props.editingContactEditing).length === 0) {
          return getBannerHeader(<FormattedMessage {...messages.newContactBanner} />);
        } else {
          return getBannerHeader(<FormattedMessage {...messages.contactEditingBanner} />);
        }
      }
      if (props.contactMode === 'merging') {
        return getBannerHeader(<FormattedMessage {...messages.contactMergeBanner} />);
      }
      return null;
  }
}

ContactHeader.propTypes = {
  editAssignedContact: PropTypes.func,
  setSearching: PropTypes.func,
};

export default Radium(ContactHeader);
