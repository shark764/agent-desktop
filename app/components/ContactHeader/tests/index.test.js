import React from 'react';
import { shallow } from 'enzyme';

import ContactHeader from '../index';

describe('<ContactHeader />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <ContactHeader
        id="mockId"
        selectedInteraction={{}}
        query={[]}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
