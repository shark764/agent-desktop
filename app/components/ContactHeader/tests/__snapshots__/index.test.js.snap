// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactHeader /> should render correctly 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "flexShrink": "0",
      "justifyContent": "center",
      "minHeight": "70px",
    }
  }
>
  <InjectIntl(Connect(ContactSearchBar))
    query={Array []}
    resultsCount={-1}
    style={
      Object {
        "paddingTop": "19px",
      }
    }
  />
  <div
    data-radium={true}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "minHeight": "65px",
        "overflowX": "auto",
        "padding": "11.5px 0",
      }
    }
  />
</div>
`;
